types/model/name.go

```diff
@@ -169,6 +175,38 @@ func ParseNameBare(s string) Name {
 	return n
 }
 
+// ParseNameFromFilepath parses a 4-part filepath as a Name. The parts are
+// expected to be in the form:
+//
+// { host } "/" { namespace } "/" { model } "/" { tag }
+func ParseNameFromFilepath(s string) (n Name) {
+	s, n.Tag = filepath.Split(filepath.Clean(s))
```

Comment:
Why not clean once up front?

Also, you can build the Name then call `IsFullyQualifed` at the end rather than check all parts by hand. It should be easier to read too.

Reply:
Instead of in each split? filepath.Split() leaves a trailing path separator. splitting it again without clean returns s and "". filepath.Clean() also doesn't clean intermediate `.` which isn't actually necessary since it's not valid anyways.

I decided to simplify this and remove the filepath functions